features/0593-json-ld-cred-attach/README.md

```diff
@@ -0,0 +1,206 @@
+
+## Motivation
+
+The Issue Credential protocol needs an attachment format to be able to exchange JSON-LD credentials with Linked Data Proofs. It is desirable to make use of specifications developed in an open standards body, such as the [Credential Manifest](https://identity.foundation/credential-manifest/) for which the attachment format is described in [RFC 0511: Credential-Manifest Attachment format](../0511-dif-cred-manifest-attach/README.md). However, the _Credential Manifest_ is not finished and ready yet, and therefore there is a need to bridge the gap between standards.
+
+## Tutorial
+
+Complete examples of messages are provided in the [reference section](#reference).
+
+## Reference
+
+### `ld-proof-vc-proposal` attachment format
+
+Format identifier: `aries/ld-proof-vc-proposal@v1.0`
+
+The credential proposal allows the holder to initiate a credential exchange without fully specifying the contents of the credential. If all properties of the credential are already known, the protocol can also be initiated with a credential request. While the credential detail requires all fields to be present, the proposal allows for fields to be omitted.
+
+The below `propose-credential` message shows an example of a credential proposal with the `@context`, `type`, `credentialSubject` and `credentialSubject` present. The issuer can then send a credential detail in a credential offer message adding extra properties required for issuance such as `issuer`, `credentialStatus.type`, `proofType`, etc..
+
+```json
+{
+  "@id": "257363ab-a3f9-4bba-ad3f-9f69501135f5",
+  "@type": "https://didcomm.org/issue-credential/%VER/propose-credential",
+  "comment": "<some comment>",
+  "formats": [
+    {
+      "attach_id": "35054fa5-112e-4ef7-93ab-7778f59eb4e3",
+      "format": "aries/vc-ld-proof@v1.0"
+    }
+  ],
+  "filter~attach": [
+    {
+      "@id": "35054fa5-112e-4ef7-93ab-7778f59eb4e3",
+      "mime-type": "application/ld+json",
+      "data": {
+        "json": {
+          "@context": [
+            "https://www.w3.org/2018/credentials/v1",
+            "https://www.w3.org/2018/credentials/examples/v1"
+          ],
+          "type": ["VerifiableCredential", "UniversityDegreeCredential"],
+          "credentialSubject": {
+            "id": "did:example:ebfeb1f712ebc6f1c276e12ec21",
+            "degree": {
+              "type": "BachelorDegree",
+              "name": "Bachelor of Science and Arts"
+            }
+          },
+          "credentialSchema": {
+            "id": "https://example.org/examples/degree.json",
+            "type": "JsonSchemaValidator2018"
+          }
+        }
+      }
+    }
+  ]
+}
+```
+
+- `credentialStatus.id` -- Credential Status may be omitted completely, but if present should contain the `type` property to indicate which mechanism to use. As the id can be closely tied to the issuance process the property is not required for the credential proposal.
```

Comment:
```suggestion
- `credentialStatus` -- Credential Status may be omitted completely, but if present should contain the `type` property to indicate which mechanism to use. As the id can be closely tied to the issuance process the property is not required for the credential proposal.
```

Reply:
Also, we should be careful with this. An existing VC parser with a disabled proof check should be able to parse this object and not raise an error. This means `credentialStatus.id` [MUST be present and MUST be a URL](https://www.w3.org/TR/vc-data-model/#status).

Reply:
Might be better to put this in an "options" object, much like the issuer vc-http-api: https://w3c-ccg.github.io/vc-http-api/#tag/Issuer

Reply:
There are more fields that have this limitation. `A verifiable credential MUST have an issuer property.`. A holder should be able to propose a credential without the `issuer` IMO.

But I totally agree on the `options` object. Seems like a better place

Reply:
I propose the `issuer` field MUST be provided in order to satisfy parsers.

However, for the purposes of this RFC, I propose the issuer MUST ignore the value provided by the Holder and instead use their own. Otherwise the entire message would need to be made non-repudiable, which is out of scope of this RFC.

Reply:
> Might be better to put this in an "options" object, much like the issuer vc-http-api: https://w3c-ccg.github.io/vc-http-api/#tag/Issuer

Note that the optionality of `credentialStatus` is not clear at the moment - I posted the question: https://github.com/w3c/vc-data-model/issues/768

Reply:
@TimoGlastra I think it makes sense to have the Holder include all fields required of a verifiable credential in the credential object (with the obvious exception of `proof`).

Allowing the Holder to define who the issuer is opens this up for more possibilities and a wider variety of use cases. Same thing goes for `issuanceDate`. Basically, I think both these fields should be REQUIRED. And again - in general, I think all fields required by the VC data model MUST be present except for `proof`